### PR TITLE
Expose asap() and a way to override the scheduling mechanism on Promise

### PR DESCRIPTION
**Pull request by vicb**
@stefanpenner This is the implementation of what we've discussed last week.

Note: The PR includes the generated JS files because it was required to test the implementation in the context of `zone.js`. Let me know if the files should be removed and generated only during the release process.

### CODE REVIEW

**Comment by vicb** [comment]
@stefanpenner any chance you take a look at the PR this week ? Thanks.

File: lib/es6-promise/promise.js [hunk 1]
```diff
@@ -155,6 +155,9 @@ Promise.all = all;
 Promise.race = race;
 Promise.resolve = Resolve;
 Promise.reject = Reject;
+// _setScheduler() and _asap() must be non-enumerable
+Object.defineProperty(Promise, '_setScheduler', {value: setScheduler});
```
Comment (stefanpenner) [inline lib/es6-promise/promise.js]: es6-promise is supposed to work in es3 browsers still. Unfortunately, no define prop :(

**Comment by stefanpenner** [comment]
@vicb thanks for the reminder, i was on vacation last week.  This looks quite reasonable, I did leave one important comment though

**Comment by vicb** [comment]
@stefanpenner thanks for the review! I have updated the PR according to your comment (and also removed the built files).

File: lib/es6-promise/asap.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 var len = 0;
 var toString = {}.toString;
 var vertxNext;
+var customSchedulerFn;
```
Comment (stefanpenner) [inline lib/es6-promise/asap.js]: how about.

```js
var schedule = defaultFlush;

...

if (len === 2) {
  schedule(flush)
}
```

**Comment by stefanpenner** [comment]
LGTM

**Comment by vicb** [comment]
@stefanpenner thanks for your help on this.

Tests are green, I have added a "2.2.0" commit updating `*.json` and dist files.

**Comment by stefanpenner** [comment]
LGTM

**Comment by stefanpenner** [comment]
@jakearchibald could we get a release good sir?

**Comment by jakearchibald** [comment]
Done!

**Comment by vicb** [comment]
:+1:

**Pull request merged**